src/tir/ir/tir_visitor_with_path.cc

```diff
@@ -199,12 +174,23 @@ void TIRVisitorWithPath::VisitStmt_(const LetStmtNode* op, ObjectPath path) {
 void TIRVisitorWithPath::VisitStmt_(const AttrStmtNode* op, ObjectPath path) {
   Visit(op->value, path->Attr("value"));
 
-  std::optional<DefContext<IterVar>> context = std::nullopt;
+  std::vector<std::variant<DefContext<IterVar>, DefContext<Var>>> context;
```

Comment:
Is this ever used? I don't see any reads from it.

Reply:
There aren't any reads from it, as it holds a scoped context manager.  On destruction, the `DefContext<T>` object removes items from `TIRVisitorWithPath::in_scope_definitions_`, and calls the `ExitDef` handler of the child class.

Also, thank you for pointing this one out.  When switching from `std::optional` to `std::vector`, I forgot to add a `while(context.size()) context.pop_back();` loop in case child classes rely on `ExitDef` being called in the reverse order from `EnterDef`.